### PR TITLE
Added setpoint generic flight controller function

### PR DESCRIPTION
**Pull request by bitcurious**
---
 
## Basic Info
 
| Info | Please fill out this column |
| ------ | ----------- |
| Ticket(s) this addresses   | |
| Primary OS tested on | Ubuntu |
| Drone platform tested on | M300  |
 
---
 
## Description of contribution in a few bullet points
 

I've added setpoint generic functionality, allowing users to customize joystick control with a single control flag.

## Motivation and Context

It will help user to have one topic to  control drone in different joystick mode using control flag.

 ## How Has This Been Tested?

## Description of documentation updates required from your changes
 

---
 
## Screenshots (if appropriate):

### CODE REVIEW

**Comment by bitcurious** [comment]
@biancabnd @vicmassy I’ve created a ROS 2 package with a script for testing. You can find the  here: [DJI RC Controller](https://github.com/bitcurious/dji_rc_controller.git). You can test the PR using this repository. It captures data from the DJI remote controller and publishes it to the setpoint generic topic with a control flag. Feel free to make any necessary changes if required.

**Comment by biancabnd** [comment]
Hi @bitcurious, 
Thank you for your contribution and apologies for the delay; I've had limited time for PR reviews lately. I will try to come back to this and review it as soon as possible.

**Comment by bitcurious** [comment]
@biancabnd No problem, take your time, also let me know if you want any additional control in [User input for controller](https://github.com/bitcurious/dji_rc_controller/blob/main/src/rc_controller.cpp#L102) like position or any different, or feel free to add that.

File: psdk_wrapper/src/modules/flight_control.cpp [hunk 1]
```diff
@@ -903,10 +903,236 @@ void
 FlightControlModule::flight_control_generic_cb(
     const sensor_msgs::msg::Joy::SharedPtr msg)
 {
-  /** @todo implemnent generic control functionality */
+  /** @author Umesh Mane
+   * Contact: umeshmane280@gmail.com
+   */
+
```
Comment (biancabnd) [inline psdk_wrapper/src/modules/flight_control.cpp:909]: Generic comments on this method. I think it is very long and makes it a bit hard to read. My suggestion would be to split it in several different methods (e.g. one method to set the horizontal axis, one for vertical, another for yaw control).

File: psdk_wrapper/src/modules/flight_control.cpp [hunk 2]
```diff
@@ -903,10 +903,236 @@ void
...
+  uint8_t HORI = (flag & 0xC0);
+  uint8_t VERT = (flag & 0x30);
+  uint8_t YAW = (flag & 0x08);
+  uint8_t FRAME = (flag & 0x06);
+  uint8_t HOLD = (flag & 0x01);
+
+  float x_cmd, y_cmd, z_cmd, yaw_cmd;
+  if (FRAME == Control::FRAME_GROUND)
+  {
+    FG = true;
```
Comment (biancabnd) [inline psdk_wrapper/src/modules/flight_control.cpp:926]: do we really need these additional variables FG, FB? Couldn't you already set the joystick mode directly here? Or is there something I am missing

File: psdk_wrapper/src/modules/flight_control.cpp [hunk 3]
```diff
@@ -903,10 +903,236 @@ void
...
+      x_cmd = y_setpoint;
+      y_cmd = x_setpoint;
+      if (HORI == Control::HORIZONTAL_VELOCITY)
+      {
+        HV = true;
+      }
+      if (HORI == Control::HORIZONTAL_POSITION)
+      {
+        HP = true;
+      }
```
Comment (biancabnd) [inline psdk_wrapper/src/modules/flight_control.cpp:940]: same comment here. Let's see if we can remove these additional variables HV, HP

**Review by biancabnd — COMMENTED** [review]
Very nice addition @bitcurious!
I left you some comments to optimize a bit the method. Let me know what you think!

**Comment by bitcurious** [comment]
@biancabnd I've updated the setpoint_generic function to make it more modular and easier to read as per your suggestions. I've also added logs to verify if the correct control mode is being selected; these can be removed after testing. The FRAME, HORI, and VERT are defined within each method to get updated values, let me know if I should leave them as is or make them global variables. Additionally, I added another file in the [dji_rc_controller](https://github.com/bitcurious/dji_rc_controller) package for selecting custom control modes as needed.

**Review by biancabnd — APPROVED** [review]

**Pull request merged**